tests/data/distance_matrix.json

```diff
@@ -0,0 +1 @@
9181262930220886], ["Rat", "Seal", 0.22339115453724195], ["Rat", "Sphenodon", 0.4065030163474816], ["Rat", "Turtle", 0.353003891348206], ["Rat", "Whale", 0.2473657985841113], ["Seal", "Bird", 0.37531874919036906], ["Seal", "Cow", 0.15451775182555724], ["Seal", "Crocodile", 0.41018137759030415], ["Seal", "Frog", 0.3681633405094492], ["Seal", "Human", 0.21002777204964668], ["Seal", "Lizard", 0.3813598336396543], ["Seal", "LngfishAf", 0.4065030163474816], ["Seal", "LngfishAu", 0.3514013266359047], ["Seal", "LngfishSA", 0.3894858370901421], ["Seal", "Mouse", 0.20540627855323415], ["Seal", "Opossum", 0.2425091673673416], ["Seal", "Platypus", 0.2600006317862731], ["Seal", "Rat", 0.22339115453724195], ["Seal", "Sphenodon", 0.3811198830850094], ["Seal", "Turtle", 0.3300953173568121], ["Seal", "Whale", 0.16879798455621745], ["Sphenodon", "Bird", 0.3597953192063457], ["Sphenodon", "Cow", 0.3777994570749373], ["Sphenodon", "Crocodile", 0.3746417339482476], ["Sphenodon", "Frog", 0.434854929508607], ["Sphenodon", "Human", 0.3894858370901421], ["Sphenodon", "Lizard", 0.37138462424904345], ["Sphenodon", "LngfishAf", 0.44083999867658924], ["Sphenodon", "LngfishAu", 0.41310277000215023], ["Sphenodon", "LngfishSA", 0.42303500819928447], ["Sphenodon", "Mouse", 0.3811198830850094], ["Sphenodon", "Opossum", 0.36956214198161425], ["Sphenodon", "Platypus", 0.4030685512731084], ["Sphenodon", "Rat", 0.4065030163474816], ["Sphenodon", "Seal", 0.3811198830850094], ["Sphenodon", "Turtle", 0.33087372946612703], ["Sphenodon", "Whale", 0.38195229160553035], ["Turtle", "Bird", 0.31393255743565457], ["Turtle", "Cow", 0.3293177123104818], ["Turtle", "Crocodile", 0.3395627858236321], ["Turtle", "Frog", 0.3613009942915812], ["Turtle", "Human", 0.35220218095652717], ["Turtle", "Lizard", 0.3426682383982826], ["Turtle", "LngfishAf", 0.38831418887971414], ["Turtle", "LngfishAu", 0.368059135466578], ["Turtle", "LngfishSA", 0.3976205688625865], ["Turtle", "Mouse", 0.33477795483657585], ["Turtle", "Opossum", 0.34027824140567847], ["Turtle", "Platypus", 0.35220218095652717], ["Turtle", "Rat", 0.353003891348206], ["Turtle", "Seal", 0.3300953173568121], ["Turtle", "Sphenodon", 0.33087372946612703], ["Turtle", "Whale", 0.3355612447250055], ["Whale", "Bird", 0.38445507479571295], ["Whale", "Cow", 0.14595671309047992], ["Whale", "Crocodile", 0.4357420182323431], ["Whale", "Frog", 0.3805364558852893], ["Whale", "Human", 0.22137140466309804], ["Whale", "Lizard", 0.4077605156884367], ["Whale", "LngfishAf", 0.40822616472178513], ["Whale", "LngfishAu", 0.3692207393580802], ["Whale", "LngfishSA", 0.40478381788695494], ["Whale", "Mouse", 0.2315249893426512], ["Whale", "Opossum", 0.24875920353669984], ["Whale", "Platypus", 0.2671130606220226], ["Whale", "Rat", 0.2473657985841113], ["Whale", "Seal", 0.16879798455621745], ["Whale", "Sphenodon", 0.38195229160553035], ["Whale", "Turtle", 0.3355612447250055]], "invalid": null, "type": "cogent3.evolve.fast_distance.DistanceMatrix", "version": "2024.12.19a2"}
```

Comment:
We should make the test case a little bit simpler. Delete this file because I don't think it's needed.

Instead, manual inspection of the results from this tree should tell you which edges give you a negative branch length.
Within a single test, set allow_negative to be true and check that the branches that you identified above do have a branch length which is less than zero. Then within the same test set allow negative equal to false and check that those branch lengths are now all zero.